File: src/services/tree-sitter/__tests__/fixtures/typescript/sample.ts
/**
 * This is a sample interface.
 */
interface MyInterface {
	id: string
	getName(): string
}

/**
 * This is a sample class implementing an interface.
 */
class MyClass implements MyInterface {
	id = "123"

	/**
	 * Gets the name.
	 */
	getName(): string {
		return "MyClass"
	}

	/**
	 * A method with parameters.
	 * @param x The first parameter.
	 * @param y The second parameter.
	 */
	processData(x: number, y: number): number {
		return x + y
	}
}

/**
 * A top-level function.
 */
function topLevelFunction() {
	const instance = new MyClass()
	console.log(instance.getName())
	console.log(instance.processData(1, 2))
}

/**
 * A nested scenario.
 */
function outerFunction(x: number) {
	/**
	 * Inner function.
	 */
	function innerFunction(y: number) {
		return x + y
	}

	class NestedClass {
		/**
		 * Nested method.
		 */
		nestedMethod() {
			return innerFunction(10)
		}
	}

	return new NestedClass().nestedMethod()
}

/**
 * A function with multiple decorators.
 */
function decorator(target: any, propertyKey: string, descriptor: PropertyDescriptor) {}

class DecoratedClass {
	@decorator
	@decorator
	methodWithDecorators() {
		return "decorated"
	}
}

/**
 * Shadowing test.
 */
const shadowed = "global"
function shadowingFunction() {
	const shadowed = "local"
	console.log(shadowed)
}

export { MyClass, topLevelFunction, outerFunction, DecoratedClass, shadowingFunction }

File: src/services/tree-sitter/__tests__/fixtures/rust/sample.rs
/// A sample struct in Rust.
pub struct RustStruct {
    pub value: i32,
}

impl RustStruct {
    /// Creates a new instance.
    pub fn new(value: i32) -> Self {
        RustStruct { value }
    }

    /// Gets the value.
    pub fn get_value(&self) -> i32 {
        self.value
    }
}

/// A sample enum.
pub enum SampleEnum {
    First,
    Second(String),
}

/// A top-level function.
pub fn outer_rust() {
    /// Nested function in Rust.
    fn inner_rust(a: i32) -> i32 {
        a * 2
    }

    mod nested_mod {
        pub fn mod_func() {
            println!("Inside nested mod");
        }
    }

    nested_mod::mod_func();
    println!("Inner: {}", inner_rust(5));
}

pub trait SampleTrait {
    fn trait_method(&self);
}

impl SampleTrait for RustStruct {
    fn trait_method(&self) {
        println!("Trait method called for value: {}", self.value);
    }
}

pub fn rust_main() {
    let instance = RustStruct::new(42);
    println!("Value: {}", instance.get_value());
    outer_rust();
    instance.trait_method();
}
